packages/jupyter/src/providers.tsx

```diff
@@ -107,3 +108,30 @@ export function ThebeLoaderAndServer({
     </ThebeBundleLoaderProvider>
   );
 }
+
+type OutputsContextType = {
+  allSafe: boolean;
+  outputsId: IdOrKey;
+};
+const OutputsContext = React.createContext<OutputsContextType | null>(null);
+
+export function useOutputsContext() {
+  const context = useContext(OutputsContext);
+  if (context === null) {
+    throw new Error('useOutputsContext must be used within a OutputsContextProvider');
+  }
```

Comment:
Hmm, what should we do if a user embeds a particular numbered output? That wouldn't pull in the outer `outputs` node.

Reply:
that is beyond the current feature set correct? i.e. currently someone can only embed the whole set of outputs?

We can think forward to that though...

Previously the `outputId` referred to the whole `output` node, now we have 

`outputs.children[output, output, ...]` referred to with `outputsId` and `outputId`s respectively.

The `useOutputContent` could allow for using an outputsID od any of its child outputId's to reference the same parent outputs node?